packages/wallet/src/services/HandlesTracker.ts

```diff
@@ -83,5 +128,19 @@ export const createHandlesTracker = ({ tip$, assetInfo$, handlePolicyIds, logger
       );
     }),
     distinctUntilChanged((a, b) => sameArrayItems(a, b, handleInfoEquals)),
+    map((handles) =>
```

Comment:
I suggest converting `hydrateHandle` to an operator similar to this:
```ts
   const hydrateHandles = (evt$: Observable<HandleInfo[]>): Observable<HandleInfo[]> => {
     // can use this to cache and not re-hydrate
     const hydratedHandles: Partial<Record<Handle, HandleInfo>> = {};
     return evt$.pipe(mergeMap((handles) => {
        ...
     }))
   }
```

```suggestion
    hydrateHandles,
```

Reply:
Fixed here: https://github.com/input-output-hk/cardano-js-sdk/pull/786/commits/2ecb53301b9c37c6727088f9fe3ac58c4d2592db.

Reply:
Nice tests with `retrySyntax` :1st_place_medal: